### ISSUE TITLE
search_issue does not return the comments field.

### ISSUE THREAD

**Issue by michaelpulis** [pycontribs/jira#1240]
### Bug summary

I am aware that a similar question has been asked before, however the response was to use the following:
`jira.search_issues(jql, fields="comment, the other fields" ... )`

I would like to retrieve ALL the fields + the comments. At runtime, my script does not know all the fields, so I cannot use the aforementioned workaround.

A workaround that I have considered is to send a small query with `maxResults:1` and without the `fields="comment"` parameter to the `search_issues(...)` function. This will allow me to obtain all of the keys of the values within the `fields` parameter. I could then join them into a single string, and append ",comment" to it, and make my future requests with this new string as the fields parameter as such:
```
all_fields = ",".join(field_names) + ",comment"
jira.search_issues(jql, fields=all_fields)
```

However, this feels like a workaround. Would it be possible to add some parameter `includeComments=True` within the `search_issues` function to do this automatically?

### Is there an existing issue for this?

- [x] I have searched the existing issues

### Jira Instance type

Jira Server or Data Center (Self-hosted)

### Jira instance version

8.14.0

### jira-python version

main

### Python Interpreter version

3.8

### Which operating systems have you used?

- [ ] Linux
- [ ] macOS
- [X] Windows

### Reproduction steps

```python
# 1. Call search_issues(...) without specifying fields paramter
# 2. Observe that the result does not include the comments field
# 3. That's the issue
```

### Stack trace

```python
none really
```

### Expected behaviour

Comments to be included, since the documentation says that omitting the fields parameter will return all the available fields.

**Comment by adehad** [comment]
excellent report, thanks.

For getting all the fields I think you should also be able to use: 

https://github.com/pycontribs/jira/blob/5ab6c29441aacc5a143f32dd01cb7512b61bbaf4/jira/client.py#L1138-L1142.


But I agree, needing to get all the fields isn't really ideal.

**Comment by michaelpulis** [comment]
I have since found what I believe to be the correct way to do it.

If you use the following:  `jira.search_issues(jql, fields="*all,comments")`, you will get all the fields and the comments too. I found this on the following page:
https://docs.atlassian.com/software/jira/docs/api/REST/8.13.13/#search-search

The link above is the documentation of the rest API, not the Python Jira Module, however it gives the desired result, as it returns all the fields and the comments too. So perhaps what was actually required is an update to the Jira Python module documentation _(https://jira.readthedocs.io/api.html?highlight=search_issues#jira.client.JIRA.search_issues)_ to mention this functionality?
